openedx/core/djangoapps/user_api/accounts/__init__.py

```diff
@@ -15,7 +15,8 @@
 
 # The minimum and maximum length for the username account field
 USERNAME_MIN_LENGTH = 2
-USERNAME_MAX_LENGTH = 30
+# Note: 30 chars is the default value for username max length, this is less than the maximum set by django.
+USERNAME_MAX_LENGTH = getattr(settings, 'OVERRIDE_USERNAME_MAX_LENGTH', 30)
```

Comment:
I would name this `USERNAME_MAX_LENGTH` instead of `OVERRIDE_...` to keep things simple.